src/StrategyEngine.cpp

```diff
@@ -0,0 +1,96 @@
+#include <unistd.h>
+#include <iostream>
+#include "StrategyEngine.h"
+
+StrategyEngine::StrategyEngine() {
+	std::cout << "Computing strategy" << std::endl;
+	return;
```

Comment:
Constructor does not return anything.
Also might be better to remove printing. Every time a strategy engine object is called it will print out this string.

Reply:
The constructor shouldnt return anything, so just leave empty perhaps.